src/Akka.Persistence.Sql/Db/AkkaDataConnection.cs

```diff
@@ -43,7 +43,7 @@ public void Dispose()
             => _connection.Dispose();
 
         public AkkaDataConnection Clone()
-            => new(_providerName, (DataConnection)_connection.Clone());
+            => throw new NotSupportedException();//_providerName, (DataConnection)_connection.Clone());
```

Comment:
We have two options here...

Option 1 - Take what I did here and treat it as Not Supported.

Option 2 - Have AkkaDataConnection take in the actual connection string as well, and then just have `Clone()` use that 

(While still marking as obsolete, need to do that still oops)

Reply:
I think we should try to preserve backward compatibility for at least a few minor versions.
I personally think option 2 with obsolete attribute for this update, then changed to obsolete attribute that throws an error on later versions. We can then remove this API on later versions.

What do you think, @Aaronontheweb?